server/src/internalClusterTest/java/org/opensearch/search/stats/StarTreeSearchStatsIT.java

```diff
@@ -0,0 +1,178 @@
+            .actionGet();
+        assertEquals(2, starTreeResponse.getHits().getTotalHits().value());
+        Terms terms = starTreeResponse.getAggregations().get("products");
+        assertEquals(2, terms.getBuckets().size());
+        Sum sum = terms.getBuckets().get(0).getAggregations().get("total_sales");
+        assertNotNull(sum);
+
+        // Query 2: This query should NOT be answered by the star-tree as it queries a non-dimension field.
+        client().prepareSearch(STARTREE_INDEX_NAME).setQuery(termQuery("non_dimension_field", "text")).execute().actionGet();
+
+        // Query 3: A standard query on the regular index.
+        client().prepareSearch(REGULAR_INDEX_NAME).setQuery(matchAllQuery()).get();
+
+        // Fetch index statistics
+        IndicesStatsResponse stats = client().admin().indices().prepareStats(STARTREE_INDEX_NAME, REGULAR_INDEX_NAME).setSearch(true).get();
+
+        // Assertions for the star-tree index
+        SearchStats.Stats starTreeIndexStats = stats.getIndex(STARTREE_INDEX_NAME).getTotal().getSearch().getTotal();
+
+        assertThat("Star-tree index should have 2 total queries", starTreeIndexStats.getQueryCount(), equalTo(2L));
+        assertThat("Star-tree index should have 1 star-tree query", starTreeIndexStats.getStarTreeQueryCount(), equalTo(1L));
+        assertThat(
+            "Star-tree query time should be non-negative",
+            starTreeIndexStats.getStarTreeQueryTimeInMillis(),
+            greaterThanOrEqualTo(0L)
+        );
+        assertThat(
+            "Star-tree query current should be non-negative",
+            starTreeIndexStats.getStarTreeQueryCurrent(),
+            greaterThanOrEqualTo(0L)
+        );
+
+        // Assertions for the regular index
+        SearchStats.Stats regularIndexStats = stats.getIndex(REGULAR_INDEX_NAME).getTotal().getSearch().getTotal();
+        assertThat("Regular index should have 1 total query", regularIndexStats.getQueryCount(), equalTo(1L));
+        assertThat("Regular index should have 0 star-tree queries", regularIndexStats.getStarTreeQueryCount(), equalTo(0L));
+        assertThat("Regular index should have 0 star-tree query time", regularIndexStats.getStarTreeQueryTimeInMillis(), equalTo(0L));
+
+        // Assertions for the total stats across both indices
+        SearchStats.Stats totalStats = stats.getTotal().getSearch().getTotal();
+        assertThat("Total query count should be 3", totalStats.getQueryCount(), equalTo(3L));
+        assertThat("Total star-tree query count should be 1", totalStats.getStarTreeQueryCount(), equalTo(1L));
+        assertThat(totalStats.getQueryTimeInMillis(), greaterThanOrEqualTo(totalStats.getStarTreeQueryTimeInMillis()));
+
+        // While not guaranteed, the time spent on the single star-tree query should be less than the total query time.
+        if (totalStats.getStarTreeQueryCount() > 0 && totalStats.getQueryCount() > totalStats.getStarTreeQueryCount()) {
+            assertThat(
```

Comment:
This can lead to flaky tests. 
Maybe assert for a non zero query time ?

Reply:
changing it to `lessThanOrEqualTo` to avoid flakiness.

Reply:
This can still be flaky due to GC pause when calculating time for Star Tree query v/s calculating time for regular queries that won't be resolved using star tree. 

IMO We can just check non zero value.

Reply:
star tree time is a subset of total query time so it will always be less, so should be okay. 
Having an upper-bound makes sense imo.